DIRHASH_STANDARD.md

```diff
@@ -0,0 +1,203 @@
+
+#### `on_cyclic_link`: `"hash reference"`
+The other option is to replace the dirhash value for the cyclic link with the hash function hexdigest of the relative path from the link to the target. The path is normalized according to the unix standard (with a forward slash `/` separating directories) and without a leading or trailing slash. 
+
+Sometimes multiple links form cycles together. Without loss of generality, cyclic links are defined as the *first occurrence of a link to a directory that has already been visited on the current branch of recursion*. The real path (or inode and device ID) of visited directories, together with the path relative to the Dirhash root, must typically be cached during traversal of the file tree to identify and resolve cyclic links. For further details, see these [examples](#cyclic-links-examples).
+
+
+### Protocol Options
+Name  | Type             | Default             | Description 
+----  | ---------------- | ------------------- | -----------
+entry_properties | Array of Strings | `["name", "data"]` | Which Directory Entry properties to consider. NOTE that `type` is a mandatory property and should not be provided
+on_cyclic_link | String, One of `"raise"` and `"hash reference"` | `"raise"` | how to handle [cyclic links](#cyclic-links).
+
+The Dirhash Protocol is designed so that the same hash should not be obtained with different Protocol Options. Subsequently, when the same hash is obtained one can be sure that the same Protocol Options were used. The options must still be provided when comparing checksums, but this removes the risk of false positives (confirmation of the checksum) due to wrong options used.
+
+
+
+## Error Conditions
+**Directory Not Accessible**: A not accessible (sub)directory results in an error unless excluded by the `match_patterns` filtering option.
+
+**File Not Accessible**: A not accessible file results in an error if the entry property `data` is used unless excluded by the `match_patterns` filtering option.
+
+**Cyclic Symbolic Links**: Presence of cyclic links, with `on_cyclic_links` set to `raise`.
+
+**Directory Empty**: No (non-empty) directory entries to hash in the Dirhash root directory, given provided Filtering Options and `include_empty_dirs` set to `false`.  
+
+
+
+## The `DIRSUM` Object
+Checksums based on the `DIRSHASH` must contain the additional configuration options to be properly validated (as was discussed in the [Introduction](#introduction)). For this purpose, the Dirhash Standard provides the `DIRSUM` object which contains all the necessary information for verification. It's structure is laid out in JSON below, with properties according [Hash Function](#hash-function-options), [Filtering](#filtering-options) and [Protocol Options](#protocol-options).
+```json
+{
+   "dirhash": "...",
```

Comment:
dirhash as in the result of the run?

Reply:
if thats the case, clearify this in the section just above

Reply:
I guess I should make a proper object description table for this but will keep it like this for now. The format is pretty clear.